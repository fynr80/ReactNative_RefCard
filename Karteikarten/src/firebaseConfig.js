
import firebase from 'firebase/compat/app';

const firebaseConfig = {
    apiKey: "AIzaSyBksswFLr1YB6en-sRf8mqSFle7bhAzSgU",
    authDomain: "swtp-812df.firebaseapp.com",
    databaseURL: "https://swtp-812df-default-rtdb.firebaseio.com",
    projectId: "swtp-812df",
    storageBucket: "swtp-812df.appspot.com",
    messagingSenderId: "44236558384",
    appId: "1:44236558384:web:2f15f6b5fb190634b71df5",
    measurementId: "G-W33L0G7D18"
};
firebase.initializeApp(firebaseConfig);
export {firebase};
