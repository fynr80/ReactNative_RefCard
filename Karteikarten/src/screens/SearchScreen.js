/**
 * @file SearchScreen of RefCard app
 * @author Kaan Eray <kaan.eray@mni.thm.de>
 */


import * as React from 'react';
import {
    View,
    Text,
    FlatList,
    TouchableOpacity,
    ActivityIndicator,
    Pressable,
    StyleSheet, Alert
} from 'react-native';
import {useEffect, useState} from "react";

import firebase from "firebase/compat";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import {SafeAreaView} from 'react-native-safe-area-context';
import Icon from "react-native-vector-icons/Ionicons";
import {Searchbar} from 'react-native-paper';
import {useNavigation} from "@react-navigation/native";
import Modal from "react-native-modal"

import stylesjs from "../style"
import SetItem from "../components/SetItem";

const SearchScreen = (props) => {
    const [items, setItems] = useState([])
    const [isLoading, setIsLoading] = useState(true)
    const [searchText, setSearchText] = useState("")
    const [fullItems, setFullItems] = useState([])
    const [orderBy, setOrderBy] = useState("none")
    const [showModal, setShowModal] = useState(false)
    const [iconName, setIconName] = useState("unsorted")

    const navigation = useNavigation()

    const category_name = props.route.params.categoriesName
    const category_id = props.route.params.id
    const categoryColor = props.route.params.categoryColor
    const path = "category/" + category_id + "/set"
    const options = ["Ascending", "Descending", "Title"]


    useEffect(() => {
        let subscriber;
        setIsLoading(true)
        subscriber = firebase.firestore()
            .collection(path)
            .onSnapshot(querySnapshot => {
                const itemList = [];
                querySnapshot.forEach(docSnapshot => {
                    itemList.push({
                        ...docSnapshot.data(),
                        id: docSnapshot.id
                    })
                })
                setItems(itemList)
                setFullItems(itemList)
                setIsLoading(false)
            })
        return () => {
            subscriber()
        }
    }, [])

    /*
    * Alert for Delete
    * */
    const button = () => {
        Alert.alert(
            'Delete Category',
            'Do you really want to delete this Category with all the Sets included?',
            [
                {
                    text: 'No', onPress: () => {
                        console.log('No Pressed')
                    }
                },
                {
                    text: 'Yes', onPress: () => {
                        //deleteSet().then(() => {
                            console.log('Collection and all Sub collections deleted');
                            deleteCategory(category_id);
                            navigation.goBack();
                        //})
                    }
                },
            ]
        );
    }



    /**
     * Searches and filters the Items to the given SearchText
     */
    const searchItem = () => {
        if (items === undefined) return
        let filteredItems
        const query = searchText.toLowerCase()
        filteredItems = fullItems
            .filter(item => {
                return item.title.toLowerCase().indexOf(query) > -1
            })
        setItems(filteredItems)
    }


    useEffect(() => {
        return searchItem()
    }, [searchText])

    useEffect(() => {
        if (orderBy === "asc" || orderBy === "desc") orderItems(compareDate)
        else if (orderBy === "title") orderItems(compareTitle)
    }, [orderBy])

    /**
     * Function to order the Items with the given criteria
     * @param func : function
     */
    const orderItems = (func) => {
        const reOrderedItems = [...items].sort((a, b) => func(a, b))
        setItems(reOrderedItems)
    }

    /**
     * Compares the date of each item in the Set and returns the result of the comparison
     * @param a : any
     * @param b : any
     * @returns {number}
     */
    function compareDate(a, b) {
        const res = b.date.seconds - a.date.seconds
        return orderBy === "asc" ? -res : res
    }

    /**
     * Compares the title of each item in the Set and returns the result of the comparison
     * @param a : any
     * @param b : any
     * @returns {number}
     */
    function compareTitle(a, b) {
        return a.title.toLowerCase().localeCompare(b.title.toLowerCase())
    }

    /**
     * Deletes the given Category with the given id
     * @param category_id : String
     */
    const deleteCategory = (category_id) => {
        deleteSet(category_id);
        firebase.firestore()
            .collection('category')
            .doc(category_id).delete()
            .then(() => {
                console.log('Deleted successfully')
            }).catch(() => {
            console.log('Deletion failed')
        })
    }

    /**
     * Deletes the theme with the associated sets
     * @param category_id : String
     */
    const deleteSet = (category_id) => {
        // Get all items
        // Loop through all Themes and delete them
        const subscriber = firebase.firestore()
            .collection('category')
            .doc(category_id)
            .collection("set")
            .get().then(res => {
            res.forEach(element => {
                deleteQuestion(element.id).then(() => console.log("Set Question successfully deleted"))
                deleteProgress(element.id).then(() => console.log("Set Progress successfully deleted"))
                element.ref.delete().then(() => console.log(""));
                console.log("Deleted Set")
            })
        })
        return () => {subscriber()};
    }

    /**
     * Deletes the questions in the associated theme
     * @param themeId : String
     */
    const deleteQuestion = async (themeId) => {
        console.log(themeId)
        await firebase.firestore()
            .collection('category').doc(category_id).collection('set').doc(themeId).collection('card').get().then(res => {
                res.forEach(element => {
                    element.ref.delete();
                    console.log("Deleted Question")
                })
            })
    }

    /**
     * Deletes the progress in the associated theme
     * @param themeId : String
     */
     const deleteProgress = async (themeId) => {
        console.log(themeId)
        await firebase.firestore()
            .collection('category').doc(category_id).collection('set').doc(themeId).collection('progress').get().then(res => {
                res.forEach(element => {
                    element.ref.delete();
                    console.log("Deleted Progress")
                })
            })
    }

    /**
     * A render method, to render each item entry in the Popup menu
     * @param item : any
     * @returns {JSX.Element}
     */
    const renderItem = ({item}) => (
        <View style={{backgroundColor: "#e6e6e6", padding: "1%"}}>
            <Pressable
                style={[styles.optionItemStyle, {
                    backgroundColor: "white",
                }]}
                onPress={() => {
                    const action = item.toString()
                    if (action === "Ascending") {
                        setIconName("sort-amount-asc")
                        setOrderBy("asc")
                    } else if (action === "Descending") {
                        setIconName("sort-amount-desc")
                        setOrderBy("desc")
                    } else if (action === "Title") {
                        setIconName("sort-alpha-asc")
                        setOrderBy("title")
                    }
                    setShowModal(!showModal)
                }
                }
            >
                <Text style={styles.optionItemText}>{item}</Text>
            </Pressable>
        </View>
    );

    /**
     * Render method for the header section of the menu
     * @returns {JSX.Element}
     */
    const renderHeader = () => (
        <View>
            <View style={{
                backgroundColor: "#e6e6e6",
                padding: "2%",
                borderTopLeftRadius: 20,
                borderTopRightRadius: 20
            }}></View>
        </View>
    )

    /**
     * Render method for the footer section of the menu
     * @returns {JSX.Element}
     */
    const renderFooter = () => (
        <View>
            <View style={{backgroundColor: "#e6e6e6", padding: "4%"}}></View>
        </View>
    )

    /**
     * Render method for empty Sets
     * @returns {JSX.Element}
     */
    const renderEmpty = () => (
        <View style={{flex: 1, alignItems: "center", justifyContent: "center"}}>
            <Text
                style={{marginTop: "10%", fontWeight: "bold", textAlign: "center", fontSize: 18}}>No Sets</Text>
        </View>
    )

    /**
     *  Renders a loading Screen, when the data is being fetched from the firebase database,
     *  otherwise a list of the current state of the items
     */
    return (
        <SafeAreaView style={stylesjs.container}>
            <View style={stylesjs.headerNew}>
                <TouchableOpacity testID="SearchBackbtn" onPress={() => {
                    navigation.goBack();
                }}>
                    <Icon name='chevron-back-sharp' size={50} color='#FD4365'/>
                </TouchableOpacity>
                <Text testID="SearchscreenTitle" style={stylesjs.headerText}>{category_name}</Text>
                <TouchableOpacity testID="SearchscreenDeletebtn" onPress={() => {
                    button();
                }}>
                    <Icon name='trash-outline' size={35} color='#FD4365'/>
                </TouchableOpacity>
            </View>
            <View style={{flex: 1, padding: "1%", flexDirection: "row", alignItems: "center"}}>
                <Searchbar
                    testID="SearchscreenSearchbar"
                    style={{borderRadius: 10, flex: 1}}
                    placeholder="Search"
                    onChangeText={(text) => setSearchText(text)}
                    value={searchText}
                />
                <TouchableOpacity
                    onPress={() => setShowModal(!showModal)}>
                    <FontAwesome name={iconName} size={24} style={{padding: "2%"}}/>
                </TouchableOpacity>
            </View>
            <View style={{flex: 9}}>
                {isLoading
                    ?
                    <View style={stylesjs.loadingContainer}>
                        <ActivityIndicator style={stylesjs.loadingStyle} size="large" color="#0377fc"/>
                        <Text>Loading</Text>
                    </View>
                    :
                    <FlatList
                        data={items}
                        keyExtractor={(item, index) => item + index}
                        renderItem={({item}) => <SetItem text={item.title}
                                                         description={item.description}
                                                         category_id={category_id}
                                                         category_name={category_name}
                                                         item_id={item.id}
                                                         categoryColor={categoryColor}
                        />}
                        ListEmptyComponent={renderEmpty}
                    />}
            </View>
            <View>
                <Modal
                    isVisible={showModal}
                    onBackdropPress={() => setShowModal(!showModal)}
                    style={{margin: 0, justifyContent: "flex-end"}}
                    backdropTransitionOutTiming={0}
                >
                    <View>
                        <FlatList
                            data={options}
                            renderItem={renderItem}
                            keyExtractor={(item, index) => item + index}
                            scrollEnabled={false}
                            ListHeaderComponent={renderHeader}
                            ListFooterComponent={renderFooter}
                        />
                    </View>
                </Modal>
            </View>
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    optionItemStyle: {
        padding: "1.5%",
        borderRadius: 20,
    },
    optionItemText: {
        fontSize: 20,
        fontWeight: "bold",
        textAlign: "center"
    }
})

export default SearchScreen
