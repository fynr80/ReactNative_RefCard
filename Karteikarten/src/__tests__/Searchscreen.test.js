import React from 'react';
import SearchScreen from "../screens/SearchScreen";
import {NavigationContainer} from "@react-navigation/native";
import {fireEvent, render} from "@testing-library/react-native";
import firebaseConfig from "../firebaseConfig";
import {toMatchImageSnapshot} from "jest-image-snapshot"
import {Alert} from 'react-native';

const puppeteer = require("puppeteer")

const mockedNavigation = jest.fn();
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigation,
            dispatch: jest.fn(),
        }),
    };
});

const route = {params: {id: "aidnawinafna", categoriesName: "MyCategory", categoryColor: '#FD4365'}}

expect.extend({toMatchImageSnapshot});

describe("SearchScreen Component", () => {
    let browser;
    jest.setTimeout(35000);

    beforeAll(async () => {
        browser = await puppeteer.launch({args: ['--no-sandbox', '--disable-setuid-sandbox']});
    })

    it('renders correctly', async () => {
        const navigate = jest.fn();
        const route = {params: {id: "aidnawinafna", categoriesName: "MyCategory", categoryColor: '#FD4365'}}
        const tree = render(
            <NavigationContainer>
                <SearchScreen route={route} navigation={{navigate}}/>
            </NavigationContainer>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("it makes a screenshot", async () => {
        const page = await browser.newPage();
        await page.goto('http://localhost:19006');
        await page.waitForSelector("#animatedComponent > div > div > div > div.css-view-1dbjc4n.r-alignItems-1awozwy.r-backgroundColor-14lw9ot.r-flex-1q4bek2.r-flexDirection-eqz5dr.r-justifyContent-1wtj0ep.r-paddingLeft-1hfyk0a.r-paddingRight-1qfoi16 > div > div > div > div:nth-child(1) > div");
        const clickSet = await page.$("#animatedComponent > div > div > div > div.css-view-1dbjc4n.r-alignItems-1awozwy.r-backgroundColor-14lw9ot.r-flex-1q4bek2.r-flexDirection-eqz5dr.r-justifyContent-1wtj0ep.r-paddingLeft-1hfyk0a.r-paddingRight-1qfoi16 > div > div > div > div:nth-child(1) > div");
        clickSet.click();
        await new Promise((r) => setTimeout(r, 5000));

        const image = await page.screenshot();

        expect(image).toMatchImageSnapshot();
    })

    it('Searchscreen Title is correct', () => {
        const {getByTestId} = render(<SearchScreen route={route} navigation={{mockedNavigation}}/>);
        const title = getByTestId("SearchscreenTitle");
        expect(title.props.children).toMatch("MyCategory");
    });

    it('Delete Category Popup opened correctly', () => {
        jest.spyOn(Alert, 'alert');
        const {getByTestId, getByText} = render(<SearchScreen route={route} navigation={{mockedNavigation}}/>);
        const button = getByTestId("SearchscreenDeletebtn");
        fireEvent.press(button);
        expect(Alert.alert).toBeCalled()
    });

    it('Searchbar Placeholder is correct', () => {
        const {getByTestId, getByText} = render(<SearchScreen route={route} navigation={{mockedNavigation}}/>);
        const placeholder1 = getByTestId("SearchscreenSearchbar");
        expect(placeholder1.props.placeholder).toMatch("Search");
    });

    /*
it('Flatlist is Empty',  () => {
    const {getByText} = render(<SearchScreen route={route} navigation={{ mockedNavigation }}/>);
    getByText("No Sets")
});
*  Loading dauerschleife deshalb finden wir die Sewts nicht. Loading l√∂schen = findet set
* */


})
