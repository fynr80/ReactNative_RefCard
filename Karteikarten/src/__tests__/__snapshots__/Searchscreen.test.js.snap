// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen Component renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
      "marginHorizontal": "2%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 2,
        "paddingRight": 2,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="SearchBackbtn"
    >
      <Text />
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
      testID="SearchscreenTitle"
    >
      MyCategory
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="SearchscreenDeletebtn"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "padding": "1%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 10,
          "elevation": 4,
          "flex": 1,
          "flexDirection": "row",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        }
      }
    >
      <View
        accessibilityLabel="search"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text />
        </View>
      </View>
      <TextInput
        accessibilityRole="search"
        accessibilityTraits="search"
        allowFontScaling={true}
        keyboardAppearance="light"
        onChangeText={[Function]}
        placeholder="Search"
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        returnKeyType="search"
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "fontSize": 18,
              "minWidth": 0,
              "paddingLeft": 8,
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            undefined,
          ]
        }
        testID="SearchscreenSearchbar"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityLabel="clear"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "opacity": 0.32,
              },
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 9,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingTop": null,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#0377fc"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      />
      <Text>
        Loading
      </Text>
    </View>
  </View>
  <View>
    <Modal
      animationType="none"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      panResponderThreshold={4}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        deviceHeight={null}
        deviceWidth={null}
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        panResponderThreshold={4}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        statusBarTranslucent={false}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View>
          <RCTScrollView
            ListFooterComponent={[Function]}
            ListHeaderComponent={[Function]}
            data={
              Array [
                "Ascending",
                "Descending",
                "Title",
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEnabled={false}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View>
              <View
                onLayout={[Function]}
              >
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#e6e6e6",
                        "borderTopLeftRadius": 20,
                        "borderTopRightRadius": 20,
                        "padding": "2%",
                      }
                    }
                  />
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#e6e6e6",
                      "padding": "1%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "borderRadius": 20,
                          "padding": "1.5%",
                        },
                        Object {
                          "backgroundColor": "white",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Ascending
                    </Text>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#e6e6e6",
                      "padding": "1%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "borderRadius": 20,
                          "padding": "1.5%",
                        },
                        Object {
                          "backgroundColor": "white",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Descending
                    </Text>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#e6e6e6",
                      "padding": "1%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "borderRadius": 20,
                          "padding": "1.5%",
                        },
                        Object {
                          "backgroundColor": "white",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Title
                    </Text>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
              >
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#e6e6e6",
                        "padding": "4%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </Modal>
  </View>
</RNCSafeAreaView>
`;
